docker.go

```diff
@@ -643,7 +643,7 @@ func (n *DockerNetwork) Remove(ctx context.Context) error {
 // DockerProvider implements the ContainerProvider interface
 type DockerProvider struct {
 	*DockerProviderOptions
-	client    *client.Client
+	client    client.APIClient
```

Comment:
Should we update the usage of client.Client elsewhere? https://github.com/testcontainers/testcontainers-go/pull/476/files#diff-b68159b6168f099a639e8c6f96d75f23333ab0f0505ee22a98268748db17ae4cR705